src/api/messageprocessors/BidMessageProcessor.ts

```diff
@@ -0,0 +1,40 @@
+import { inject, named } from 'inversify';
+import { validate } from '../../core/api/Validate';
+import { Logger as LoggerType } from '../../core/Logger';
+import { Types, Core, Targets } from '../../constants';
+import { MessageProcessorInterface } from './MessageProcessorInterface';
+import { BidFactory } from '../factories/BidFactory';
+import { Bid } from '../models/Bid';
+import { BidMessage } from '../messages/BidMessage';
+import { BidService } from '../services/BidService';
+import { ListingItemService } from '../services/ListingItemService';
+import { NotFoundException } from '../exceptions/NotFoundException';
+
+export class BidMessageProcessor implements MessageProcessorInterface {
+
+    public log: LoggerType;
+
+    constructor(
+        @inject(Types.Factory) @named(Targets.Factory.BidFactory) private bidFactory: BidFactory,
+        @inject(Types.Service) @named(Targets.Service.BidService) private bidService: BidService,
+        @inject(Types.Service) @named(Targets.Service.ListingItemService) private listingItemService: ListingItemService,
+        @inject(Types.Core) @named(Core.Logger) public Logger: typeof LoggerType
+    ) {
+        this.log = new Logger(__filename);
+    }
+
+    @validate()
+    public async process(message: BidMessage ): Promise<Bid> {
+        // find listingItem by hash
+        const listingItem = await this.listingItemService.findOneByHash(message.item);
+        // if listingItem not found
+        if (listingItem === null) {
+            this.log.warn(`ListingItem with the hash=${message.item} was not found!`);
+            throw new NotFoundException(message.item);
+        } else {
+            message.item = listingItem.id;
+            const bidData = this.bidFactory.get(message);
```

Comment:
can you switch this around? if you set listingItem.id to message.item then you're not really converting BidMessage to Bid, but instead youre converting BidMessage where item (hash) has been replaced with listingItem.id to Bid, which is not expected, confusing and the unexpected behaviour is not documented.

so, I think..
let bidData = this.bidFactory.get(message);
bidData.listing_item_id = listingItem.id;  // setting the relation
.. would be better